debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/YbExporterConsumer.java

```diff
@@ -141,11 +140,17 @@ else if (exporterRole.equals(TARGET_DB_EXPORTER_ROLE)){
     }
 
     private void checkForSwitchOperationAndHandle(String operation){
-        File switchTriggerFile = new File(Path.of(triggersDir, operation).toString());
-        if (!switchTriggerFile.exists()){
+        try {
+            Boolean exists = exportStatus.checkIfSwitchOperationRequested(operation);
+            if (!exists) {
+                return;
+            }
+        } catch (SQLException e) {
```

Comment:
I thought generally the operation/queries inside `checkIfSwitchOperationRequested()` func should not fail. Since they are tried and tested. If they do, we can retry the check.

But now i think erroring out is also fine, let me do the change

Reply:
Yeah, my concern was: if you ignore an exception and simply retry, you need to have some limits in place, so that after a certain time/no. of retries, the error is thrown up to the user.. It's simpler to just throw an error now, since voyager is anyway resumable..